docs/DeveloperGuide.md

```diff
@@ -607,7 +605,165 @@ testers are expected to do more *exploratory* testing.
    1. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is larger than the list size)<br>
       Expected: Similar to previous.
 
-1. _{ more test cases …​ }_
+### Adding a delivery
+
+1. Adding a delivery with existing supplier
+
+    1. Prerequisites: List all suppliers using the `list -s` command. Ensure there is at least 1 supplier.
+
+    1. Test case: `add -d on/18-01-2023 15:00 s/1 pro/bread q/500 g c/5.50`<br>
+       Expected: A delivery paired to the first supplier is added. Details of the delivery shown in the status message.
+
+2. Adding a delivery with invalid supplier
+    1. Test case: `add -d on/18-01-2023 15:00 s/0 pro/bread q/500 g c/5.50`<br>
+       Expected: No delivery is added. Error message indicating that SUPPLIER_INDEX should be a positive number greater than 0 and smaller than total number of suppliers shown in the status message.
+
+3. Adding a delivery with invalid units for QUANTITY
+   1. Prerequisites: List all suppliers using the `list -s` command. Ensure there is at least 1 supplier.
+   
+   1. Test case: `add -d on/18-01-2023 15:00 s/1 pro/bread q/500 pounds c/105.50`<br>
+      Expected: No delivery is added. Error message indicating that QUANTITY should be a positive number followed by a space and valid unit shown in the status message. 
+
+4. Adding a delivery with invalid parameter values
+    1. Prerequisites: List all suppliers using the `list -s` command. Ensure there is at least 1 supplier.
+
+    1. Test case: `add -d on/18-01-2023 15:00 s/1 pro/###@@ q/500 pounds c/105.50`<br>
+       Expected: No delivery is added. Error message indicating that PRODUCT should be alphanumeric shown in the status message.
+
+### Listing all deliveries
+
+1. Listing all deliveries
+    1. Test case: `list -d`<br>
+       Expected: All added deliveries shown. Success message shown in the status message.
+
+### Marking a delivery
+
+1. Marking a delivery as CANCELLED
+
+    1. Prerequisites: List all deliveries using the `list -d` command. Ensure there is at least 1 delivery with either PENDING or DELIVERED status.
```

Comment:
Should this be ensure that the first delivery has a PENDING or DELIVERED status?

Reply:
Yup i have included the check for first delivery